### PR TITLE
Test Complete. Sorry for the delay I was busy and I had to enter the ful...

### PR DESCRIPTION
**Pull request by CalienteRaymond**
Test Complete. Sorry for the delay I was busy and I had to enter the full stack with codeception, ruckus, ZF1 Rest (not that fresh with ZF), and fix the test setup a little bit

It’s not as good as I want but I’m clearly running out of time and did that overtime. I would have like to develop a resend button or work with the actual twill API, or manage an external list of keywords, or save the completion confirmation, and use karma. Maybe realtime is it nevertheless match your expectation and I pass by again in Bcn.

- Back
	* TackAction Controller + model Added
	* TwilioRequest Controller + model Added -> To simulate incoming Twilio request with proper Args. Will create correspondant TaskActions.

Ruckus DB sample and Db Schema updates.

Relevant Functional tests added, no unit test

- Front:
	* angular page update
	* added modal page for task action list
	* Dep included, see bower.json

Flaws:
* Miss functional test for already answered Tasks
* Miss proper sorting of tasks for functional rules
* Dirty Regex
* Bootstrap.css conflicts with fuelux.css and not sure this extension is indispensable
* sorry, no karma tests.

### CODE REVIEW

File: library/Vreasy/Models/Task.php [hunk 1]
```diff
@@ -13,6 +13,10 @@ class Task extends Base
     protected $assigned_phone;
     protected $created_at;
     protected $updated_at;
+    protected $description;             // What is the task dedicated to ?
+    protected $state;                   // Pending, accepted, refused (exclusive)
+    protected $complete_claimed;        // Has the provider claimed he finish the job ?
```
Comment (brutuscat) [inline library/Vreasy/Models/Task.php]: @CalienteRaymond why haven't you included these as states too? Could you think of ways to improve this?

**Comment by brutuscat** [comment]
@CalienteRaymond thank you for doing the test assignment. I've done a couple of comments and I'm waiting for some more review from the team.

**Comment by brutuscat** [comment]
@CalienteRaymond could you write me an email with your personal details so we can find you in our records? mauro at vreasy dot com

File: library/Vreasy/Models/Task.php (hunk 1 above)
Comment (CalienteRaymond) [inline library/Vreasy/Models/Task.php]: To me complete_claimed and complete_confirmed were booleans representing a different level of information; a task could be not formally accepted while claimed complete or observed as complete. state should be reserved for twilio message interpretation.

But it is a functional choice primarly, it has to be discussed. Or course we could consider a sequential state for the task from pending to confirmed complete.

Alexandre Damiron
Caliente Raymond SAS
alexandre.damiron@caliente-raymond.com
(+33) 06 26 33 38 41

Comment (mwpicard) [inline library/Vreasy/Models/Task.php]: Salut Alexandre,.

notre CTO a aime' ton boulot.

Alors nous on voudrais te faire une offre...

veux tu en parler par skype?

ciao

Martin Picard <http://www.linkedin.com/in/martinpicard>
Founder

Vreasy Corporation <http://www.vreasy.com>
*HQ:*
7800 N MoPac,
Austin, TX 78759
*R&D  - MAP <http://bit.ly/VreasyMAP>*
57 Atlantida
Barcelona 08003
Tel: +34606551138
Email: martin@vreasy.com
Skype: mpskyper
<https://docs.google.com/document/d/1vkKIwyvi786v7W9L-Pzh-bXr8Agrzo9hFZNePVaS49c/pub>

*Vreasy.com* wants YOU for our Barcelona R&D offices. Join our talented
international team and work minutes from the beach.  Be inspired to
achieve.
<https://docs.google.com/document/d/1vkKIwyvi786v7W9L-Pzh-bXr8Agrzo9hFZNePVaS49c/pub>Check
out our Jobs Board
<https://docs.google.com/document/d/1vkKIwyvi786v7W9L-Pzh-bXr8Agrzo9hFZNePVaS49c/pub>

Comment (CalienteRaymond) [inline library/Vreasy/Models/Task.php]: Salut Martin, j’en suis ravi. Le 8 mai est férié en France, obligations de famille, et lundi je suis très très chargé.

Mardi, 13H ?

Sinon, n’oublie pas que je prévois un passage à barcelone entre le 18 et le 30, je pense qu’on aura le plus gros de nos discussions en direct.

A bientôt

Alexandre Damiron
Caliente Raymond SAS
alexandre.damiron@caliente-raymond.com
(+33) 06 26 33 38 41

**Comment by CalienteRaymond** [comment]
Hey,

Just look at that interesting stuff, I love it! I think you're gonna like it too, take a look here <http://prantobylte.heifersgrill.com/aexaqf>

alexandre.damiron@caliente-raymond.com

**Pull request closed without merging**